### PR TITLE
Fsr OR GeoFit

### PR DESCRIPTION
**Pull request by irenedutta23**
If FSR correction is applied, avoid GeoFit correction.

### CODE REVIEW

File: tests/hmm/hmumu_utils.py [hunk 1]
```diff
@@ -1440,6 +1445,8 @@ def get_selected_muons(
             mu_phi = NUMPY_LIB.array(mu_phi)
             mu_mass = NUMPY_LIB.array(mu_mass)
             mu_iso = NUMPY_LIB.array(mu_iso)
+
+        is_not_fsr = (muons.pt == mu_pt)
```
Comment (nanlu06) [inline tests/hmm/hmumu_utils.py:1449]: This is for CPU part. maybe a line for the GPU part of the code can be added too?

Comment (nanlu06) [inline tests/hmm/hmumu_utils.py:1449]: I got an error message when running this:

in get_selected_muons
    "is_not_fsr": is_not_fsr,
UnboundLocalError: local variable 'is_not_fsr' referenced before assignment

Comment (irenedutta23) [reply tests/hmm/hmumu_utils.py:1449]: Yes, that was a bug when not using the option `do_fsr`. It's fixed now.

**Comment by irenedutta23** [comment]
Datacards are in `/storage/user/idutta/hmm/PR111`

**Comment by nanlu06** [comment]
Hi @irenedutta23 about line   1449      is_not_fsr = (muons.pt == mu_pt) I was wondering if we can do this inside correct_muon_with_fsr() function? and check if out_muons_fsrPhotonIdx is negative, is_not_fsr = 1 ?

I was a bit concerned about muons.pt == mu_pt will work well due to precision issues.

**Comment by irenedutta23** [comment]
Hi @nanlu06 , thanks for your comments. I updated the code accordingly and I'm running new jobs now. I will let you know when they are ready.

**Pull request merged**